### PR TITLE
Unified heap and Heap size settings

### PR DESCRIPTION
**Pull request by valdisxp1**
## Made the heap unified
- `LargeAllocator` now uses superblocks (superblock = multiple consecutive blocks) within the small heap.
- `Heap_Recycle` now delegates to the specific sweepers per block basis.
-  Free blocks are no longer swept.
-  Both `Allocator` and `LargeAllocator` ask a common `BlockAllocator` for free blocks. 

## Added heap size settings
- Change the minimum heap size by setting the `SCALANATIVE_MIN_HEAP_SIZE` environment variable.
- Change the maximum heap size by setting the `SCALANATIVE_MAX_HEAP_SIZE` environment variable.
- They can go as low as 1 MB and as high as your system memory up to about 512 GB.
- The size is specified in bytes. Suffixes can be used to specify  kilobytes(`k`), megabytes(`m`) or gigabytes(`g`). Both upper and lower case is accepted.

Solves #1358 and #1312 

The block allocator does not have a noticeable impact on benchmarks. They get 0.19% faster at 50 percentile.
Full report:
https://github.com/valdisxp1/scala-native-benchmark-results/blob/gh-pages/reports/summary_20181025_144930_scala-native-0.3.9-SNAPSHOT%40log-gc-times.e5ceb7bd.-gc_vs_scala-native-0.3.9-SNAPSHOT%40block-allocator-gc-gc/Readme.md

### CODE REVIEW

File: nativelib/src/main/resources/gc/immix/metadata/BlockMeta.h [hunk 1]
```diff
@@ -9,45 +11,90 @@
...
+    union {
+        struct {
+            uint8_t flags;
+            int8_t first;
+        } simple;
+        struct {
+            uint8_t flags;
+            int32_t size : BLOCK_COUNT_BITS;
+        } superblock;
+    } block;
```
Comment (densh) [inline nativelib/src/main/resources/gc/immix/metadata/BlockMeta.h]: BlockMeta struct stays 64-bit sized in the end, which is great. https://godbolt.org/z/9GXIan

**Comment by densh** [comment]
Needs a rebase due to conflicting changes in gc logging PR, otherwise looks good to me!

**Comment by valdisxp1** [comment]
Rebased.
Thanks @ekrich for pointing out the warning on 32 bit compilation,

**Comment by densh** [comment]
LGTM, well done @valdisxp1 ! 👍

**Pull request merged**